src/hor2eq.jl

```diff
@@ -0,0 +1,137 @@
+
+### Explanation ###
+
+This is a function to calculate equatorial (ra,dec) coordinates from
+horizon (alt,az) coords. It is accurate to about 1 arcsecond or better.
+It performs precession, nutation, aberration, and refraction corrections.
+
+### Arguments ###
+
+* `alt`: altitude of horizon coords, in degrees
+* `az`: azimuth angle measured East from North (unless ws is `true`), in degrees
+* `jd`: julian date
+* `ws` (optional boolean keyword): set this to `true` to get the azimuth measured
+  westward from south
+* `B1950` (optional boolean keyword): Set this to `true` if the ra and dec
+  are specified in B1950 (FK4 coordinates) instead of J2000 (FK5). This is `false` by
+  default
+* `precession` (optional boolean keyword): set this to `false` for no precession,
+  `true` by default
+* `nutate` (optional boolean keyword): set this to `false` for no nutation,
+  `true` by default
+* `aberration` (optional boolean keyword): set this to `false` for no aberration
+  correction, `true` by default
+* `refract` (optional boolean keyword: set this to `false` for no refraction
+  correction, `true` by default
+* `lat` (optional keyword): north geodetic latitude of location, in degrees. Default
+  is NaN
+* `lon` (optional keyword): AST longitude of location, in degrees. You can specify west
+  longitude with a negative sign. Dafault is NaN.
+* `altitude` (optional keyword): the altitude of the observing location, in meters.
+  It it zero by default
+* `pressure` (optional keyword): the pressure at the observing location, in millibars.
+  Default is NaN
+* `temperature` (optional keyword): the temperature at the observing location, in Kelvins.
+  Default is NaN
+* `obsname` (optional keyword): set this to a valid observatory name to
+  be used by the observatory type in [types](@ref), which will return
+  the latitude and longitude to be used by this program. Default is NaN.
+
+### Output ###
+
+* `ra`: right ascension of object, in degrees (FK5)
+* `dec`: declination of the object, in degrees (FK5)
+* `ha`: hour angle, in degrees
+
+### Example ###
+
+You are at Kitt Peak National Observatory, looking at a star at azimuth
+angle 264d 55m 06s and elevation 37d 54m 41s (in the visible). Today is Dec 25, 2041
+and the local time is 10 PM precisely. What is the right ascension and declination
+(J2000) of the star you're looking at? The temperature here is about 0 Celsius,
+and the pressure is 781 millibars. The Julian date for this time is 2466879.7083333
+
+```jldoctest
+julia> ra_o, dec_o = hor2eq(ten(37,54,41), ten(264,55,06), 2466879.7083333, obsname="kpno",
+                            pressure = 711, temperature = 273)
+(3.3222851503189124, 15.190605763758745, 3.640795457403172)
+
+julia> adstring(ra_o, dec_o)
+" 00 13 17.3  +15 11 26"
+```
+
+### Notes ###
+
+Code of this function is based on IDL Astronomy User's Library.
+"""
+hor2eq(alt::Real, az::Real, jd::Real; ws::Bool=false, B1950::Bool=false,
```

Comment:
In `_hor2eq`, put all floating point arguments one after the other, so that you can promote all of them together:
```julia
_hor2eq(promote(float(alt), float(az), float(jd), float(lat), float(lon), float(altitude),
                float(temperature), float(pressure))..., ws,
        B1950, precession, nutate, aberration, refract,
        obsname)
```